### PR TITLE
ADAP-366: View Rerun Bug

### LINKED ISSUE #365: [ADAP-366] [Bug] Issue with table materialization
### Is this a regression in a recent version of dbt-redshift?

- [X] I believe this is a regression in dbt-redshift functionality
- [X] I have searched the existing issues, and I could not find an existing issue for this regression

### Current Behavior

when running dbt on-redshift on 1.5.0.b2, the backup relation step of the materialization seems to fail. 

```
[0m21:24:23.321701 [debug] [Thread-1  ]: Using redshift connection "model.dbt_project_evaluator.stg_exposure_relationships"
[0m21:24:23.322455 [debug] [Thread-1  ]: On model.dbt_project_evaluator.stg_exposure_relationships: alter table "ci"."dbt_dconnors"."stg_exposure_relationships" rename to "stg_exposure_relationships__dbt_backup"
[0m21:24:23.376670 [debug] [Thread-1  ]: Redshift adapter: Redshift error: {'S': 'ERROR', 'C': '42P01', 'M': 'relation "dbt_dconnors.stg_exposure_relationships" does not exist', 'F': '../src/pg/src/backend/catalog/namespace.c', 'L': '262', 'R': 'LocalRangeVarGetRelid'}
[0m21:24:23.377097 [debug] [Thread-1  ]: On model.dbt_project_evaluator.stg_exposure_relationships: ROLLBACK
```

Seems to be unable to find a relation in the `load_cached_relation` step, and throws a `LocalRangeVarGetRelid` error. 

### Expected/Previous Behavior

In 1.4.4 this same project runs without error!

### Steps To Reproduce

1. run a table model in v1.4-latest
2. switch to 1.5.0b2
3. run it again
4. observe the error

### Relevant log output

_No response_

### Environment

```markdown
- OS: Mac
- Python: Python 3.9.13
- dbt-core (working version): 1.4.4
- dbt-redshift (working version): 1.4.0
- dbt-core (regression version): 1.5.0-b3
- dbt-redshift (regression version): 1.5.0-b2
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by mikealfare**
resolves #365

### Description

- Update `RedshiftAdapter` to include link_cache_relation methods that were inadvertently dropped when migrating from inheriting from `PostgresAdapter` to inheriting directly from `SQLAdapter`
- Add test case to catch this error in the future

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-redshift/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-redshift/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: dbt/adapters/redshift/impl.py [hunk 1]
```diff
@@ -99,3 +104,53 @@ def valid_incremental_strategies(self):
...
+
+    def _link_cached_relations(self, manifest):
+        schemas = set(
+            relation.schema.lower()
+            for relation in self._get_cache_schemas(manifest)
+            if self.verify_database(relation.database) == ""
+        )
+        self._link_cached_database_relations(schemas)
+
+    def _relations_cache_for_schemas(self, manifest, cache_schemas=None):
```
Comment (VersusFacit) [inline dbt/adapters/redshift/impl.py:140]: for reference only: `dbt_postgres` has its own implementation of these methods at `core/dbt/adapters/postgres/imply.py`

Comment (mikealfare) [reply dbt/adapters/redshift/impl.py:140]: Correct, those are the ones that we accidentally dropped when dropping the inheritance on `PostgresAdapter`.

**Review by VersusFacit — APPROVED** [review]
Looks generally good to me. Just one question that's more curiosity than a blocker.

File: dbt/adapters/redshift/impl.py [hunk 2]
```diff
@@ -99,3 +104,53 @@ def valid_incremental_strategies(self):
...
+        super()._relations_cache_for_schemas(manifest, cache_schemas)
+        self._link_cached_relations(manifest)
+
+    # avoid non-implemented abstract methods warning
+    # make it clear what needs to be implemented while still raising the error in super()
+    # we can update these with Redshift-specific messages if needed
+    @property
+    def python_submission_helpers(self) -> Dict[str, Type[PythonJobHelper]]:
```
Comment (VersusFacit) [inline dbt/adapters/redshift/impl.py:148]: These are technically still abstract methods, eh? Since they super to unimplemented ones under the hood?

**Pull request merged**